src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -107,7 +107,7 @@ int MPIDU_shm_seg_alloc(size_t len, void **ptr)
             serialized_hnd_size = strlen(serialized_hnd);
             MPIR_Assert(serialized_hnd_size < MPIDU_SHM_MAX_FNAME_LEN);
 
-            MPIDU_Init_shm_put(serialized_hnd, MPIDU_SHM_MAX_FNAME_LEN);
+            MPIDU_Init_shm_put(serialized_hnd, serialized_hnd_size);
```

Comment:
Yes, we need +1 to account the 0-terminator. We also need remove the `MPIR_Assert`. There is an assert inside `MPIDU_Init_shm_put` anyway.

Reply:
OK, will fix.